python/django/security/audit/xss/global-autoescape-off.yaml

```diff
@@ -0,0 +1,21 @@
+rules:
+- id: global-autoescape-off
+  message: |
+    Autoescape is globally disbaled for this Django application. If you are
+    rendering any web pages, this exposes your application to cross-site
+    scripting (XSS) vulnerabilities. Remove 'autoescape: False' or set it
+    to 'True'.
+  metadata:
+    cwe: "CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')"
+    owasp: 'A7: Cross-site Scripting (XSS)'
+    references:
+    - https://docs.djangoproject.com/en/3.1/ref/settings/#templates
+    - https://docs.djangoproject.com/en/3.1/topics/templates/#django.template.backends.django.DjangoTemplates
+  languages:
+  - python
+  severity: WARNING
+  pattern: |
+    {..., 'BACKEND': ..., 'OPTIONS': {..., 'autoescape': False, ...}, ...}
```

Comment:
can backend be something other than 'django.template.backends.django.DjangoTemplates' in this case?

Reply:
Yes, it can be "Jinja2" which uses Jinja2 as the template engine. However, if my read of the docs is correct, the 'OPTIONS' can still include `'autoescape': False`.

See a few lines into https://docs.djangoproject.com/en/3.1/topics/templates/#django.template.backends.jinja2.Jinja2 for where I'm getting this info. Hopefully my read is correct.

Reply:
ah yes makes sense!